nucypher/acumen/perception.py

```diff
@@ -168,8 +218,91 @@ def abridged_state_details(state):
                 "updated": state.updated.rfc2822(),
                 }
 
-    def mark_as(self, label: Exception, node: "Teacher"):
-        self._marked[label].append(node)
-
-        if self._nodes.get(node):
-            del self._nodes[node]
+    def __reset_node_label_tracking(self, node: "Teacher"):
+        for strategy in self.__pruning_strategies:
+            strategy(node=node, reset=True)
+
+    def register_node_pruning_strategy(self,
```

Comment:
Is it called from anywhere?

Reply:
Not sure yet. This essentially stems from the potential need to allow arbitrary pruning strategies to be set dynamically while still having a path to reset it's internal state.

Reply:
I don't think there's an immediate need for that. The user can't change the node's behavior in runtime (would be cool to process SIGHUP for that though).